### ISSUE TITLE
[firebase_message] onTokenRefresh being called every time the app loads

### ISSUE THREAD

**Issue by thanhbinh84** [FirebaseExtended/flutterfire#1824]
**Describe the bug**
onTokenRefresh being called every time the app loads even the tokens are the same

**To Reproduce**
`_firebaseMessaging.onTokenRefresh.listen((token) => _saveToken(token)); // save token when it was refreshed`

**Expected behavior**
only fire when token is refreshed 

**Additional context**
firebase_messaging: ^6.0.3

**Comment by LukaGiorgadze** [comment]
Still not fixed.

**Comment by thanhbinh84** [comment]
> Still not fixed.

You can workaround by storing token local and compare to ignore in case it is the same

**Comment by LukaGiorgadze** [comment]
> > Still not fixed.
> 
> You can workaround by storing token local and compare to ignore in case it is the same

Yeah, this is exactly what did it :) thanks!

**Comment by xJon** [comment]
Is there any upcoming fix? This is quite disappointing.

**Comment by LukaGiorgadze** [comment]
@xJon You can do workaround, something like this:

```
Stream<String> fcmStream = instance.onTokenRefresh.distinct();

fcmStream.listen((token) async {
  SharedPreferences prefs = await SharedPreferences.getInstance();
  String fcmToken = prefs.getString('fcm_token');

  if (fcmToken == token) return;

  // ...
  // create new token below and save it in SharedPreferences
  // ...
});
```

**Comment by tahlers93** [comment]
Still not fixed.

**Comment by xJon** [comment]
Doesn't seem to happen anymore with `8.0.0-dev.14`.

**Comment by russellwheatley** [comment]
hey @thanhbinh84, can you confirm this behaviour has been fixed in the latest version: `9.1.2`? Thanks.

**Comment by russellwheatley** [comment]
I'm going to close this as a user has provided feedback confirming it was fixed in a previous version `8.0.0-dev.14`. Will reopen if this is not the case.
